artifacts/definitions/Windows/System/Services.yaml

```diff
@@ -5,21 +5,37 @@ description: |
 parameters:
   - name: servicesKeyGlob
     default: HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\
-
+  - name: Calculate_Hash
+    default: N
+    type: bool
+  - name: CertificateInfo
+    default: N
+    type: bool  
+    
 sources:
   - precondition: |
       SELECT OS From info() where OS = 'windows'
 
     queries:
       - |
-        SELECT State, Name, DisplayName, Status,
+        LET service <= SELECT State, Name, DisplayName, Status,
                ProcessId as Pid, ExitCode, StartMode,
                PathName, ServiceType, StartName as UserAccount,
                {
-                  SELECT timestamp(epoch=Mtime.Sec) as Created
-                  FROM stat(filename=servicesKeyGlob + Name, accessor='reg')
+                 SELECT timestamp(epoch=Mtime.Sec) as Created
+                 FROM stat(filename=servicesKeyGlob + Name, accessor='reg')
                } AS Created,
                {
                  SELECT ServiceDll FROM read_reg_key(globs=servicesKeyGlob + Name + "\\Parameters")
-               } AS ServiceDll
+               } AS ServiceDll,
+               parse_string_with_regex(regex=
+                 ['^\\"(?P<AbsolutePath>[^"]+)','(?P<AbsolutePath>^[^"]+)[ ][-~/]+','(?P<AbsolutePath>^[^ ]+$)'], 
+                 string=PathName) as Parsed
```

Comment:
Since there is really only one field here this might look a bit better:

```
          parse_string_with_regex(regex=
                 ['^\\"(?P<AbsolutePath>[^"]+)','(?P<AbsolutePath>^[^"]+)[ ][-~/]+','(?P<AbsolutePath>^[^ ]+$)'], 
                 string=PathName).AbsolutePath as AbsolutePath
```

2. I think the `\\` might not be necessary ` ['^"(?P<AbsolutePath>[^"]+)', `

3. I dont know if it makes sense to require `[-~/]+` after the command - this will not match a lot of commands (like `sc create` or `net user`) maybe drop it?

Reply:
1. Great, that looks better indeed, I changed it.

2. The `\\` is sometimes necessary since the value PathName can start with `\\` : 

![image](https://user-images.githubusercontent.com/11965973/77330852-342d8180-6d20-11ea-948f-b6024f3250a1.png)

3. I can remove the `~` since this would not occur very often, but the rest are required to extract the absolute path from the following services entries in registry:

`C:\\Windows\\System32\\svchost.exe` -k wsappx
`C:\\Windows\\system32\\msiexec.exe` /V

Reply:
2. I dont think it actually starts with `\` - it is just showing you that because it starts with `"` and it is inside a string so it needs to be quoted. So the case you showed above starts with `"` actually.

3. If you only captured until the space it would still work and would get commands with args that do not contain flags (like `sc create`). 
```
'(?P<AbsolutePath>^[^"]+) '
```

Reply:
2. Check, I removed it and it works!

3. Also works, thanks.